lib/puppet/util/windows/file.rb

```diff
@@ -89,10 +89,31 @@ def symlink(target, symlink)
   end
   module_function :symlink
 
+
+  def file_exist(file_name)
+    path = file_name.to_s
+    attempts = 10
```

Comment:
Yes.  To avoid circular references a simple max depth was set.  Do you think it should be higher?